src/Core/Services/Implementations/PolicyService.cs

```diff
@@ -132,6 +133,17 @@ public class PolicyService : IPolicyService
         await _eventService.LogPolicyEventAsync(policy, Enums.EventType.Policy_Updated);
     }
 
+    public async Task<ICollection<OrganizationUserPolicyDetails>> GetPoliciesApplicableToUserAsync(Guid userId, PolicyType policyType, OrganizationUserType minUserType = OrganizationUserType.User, OrganizationUserStatusType minStatus = OrganizationUserStatusType.Accepted)
+    {
+        var organizationUserPolicyDetails = await _organizationUserRepository.GetByUserIdWithPolicyDetailsAsync(userId);
+        return organizationUserPolicyDetails.Where(o =>
+                     o.PolicyType == policyType &&
+                     o.PolicyEnabled &&
+                     o.OrganizationUserType >= minUserType &&
```

Comment:
The caller should not need to know whether (as a matter of business logic) this `PolicyType` applies to this `OrganizationUserType`. That's the job of this method, and that's what we're trying to encapsulate here. (e.g. if we change how policies work, we only have to update this method and not every calling location.)

We should basically have a switch statement here where it determines the `minUserType` based on the `PolicyType` specified in the arguments. Then you don't need `minUserType` as a parameter.

(`minStatus` is a bit different because that does actually depend on the circumstances and we need the caller to tell us that.)

Reply:
Makes perfect sense. I've delegated that solely to `PolicyService` with a private method `GetMinimumOrganizationUserTypeForPolicyType`